File: 344.reverse-string.js
/*
 * @lc app=leetcode id=344 lang=javascript
 *
 * [344] Reverse String
 */

// @lc code=start
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
  var b = s.length - 1;
  for (var a = 0; a < parseInt(s.length / 2); a++) {
    var temp = s[a];
    s[a] = s[b];
    s[b] = temp;
    b--;
  }
};
// @lc code=end

File: src/344.reverse-string.cpp
/*
 * @lc app=leetcode id=344 lang=cpp
 *
 * [344] Reverse String
 */
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
 public:
  void reverseString(vector<char>& s) {
    // 1-line solution
    // reverse(s.begin(), s.end());

    for (int i = 0; i < s.size() / 2; i++) {
      int temp = s[i];
      s[i] = s[s.size() - 1 - i];
      s[s.size() - 1 - i] = temp;
    }
  }
};
// @lc code=end
